src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java

```diff
@@ -160,15 +165,23 @@ public EC2AbstractSlave(String name, String instanceId, String templateDescripti
         this.idleTerminationMinutes = idleTerminationMinutes;
         this.tags = tags;
         this.usePrivateDnsName = connectionStrategy == ConnectionStrategy.PRIVATE_DNS;
-        this.useDedicatedTenancy = useDedicatedTenancy;
+        this.useDedicatedTenancy = tenancy == Tenancy.Dedicated;
         this.cloudName = cloudName;
         this.launchTimeout = launchTimeout;
         this.amiType = amiType;
         this.maxTotalUses = maxTotalUses;
+        this.tenancy = tenancy != null ? tenancy : Tenancy.Default;
         readResolve();
         fetchLiveInstanceData(true);
     }
 
+    @Deprecated
+    public EC2AbstractSlave(String name, String instanceId, String templateDescription, String remoteFS, int numExecutors, Mode mode, String labelString, ComputerLauncher launcher, RetentionStrategy<EC2Computer> retentionStrategy, String initScript, String tmpDir, List<? extends NodeProperty<?>> nodeProperties, String remoteAdmin, String jvmopts, boolean stopOnTerminate, String idleTerminationMinutes, List<EC2Tag> tags, String cloudName, boolean useDedicatedTenancy, int launchTimeout, AMITypeData amiType, ConnectionStrategy connectionStrategy, int maxTotalUses)
+            throws FormException, IOException {
+
+        this(name, instanceId, templateDescription, remoteFS, numExecutors, mode, labelString, launcher, retentionStrategy, initScript, tmpDir, nodeProperties, remoteAdmin, jvmopts, stopOnTerminate, idleTerminationMinutes, tags, cloudName, launchTimeout, amiType, connectionStrategy, maxTotalUses, Tenancy.Default);
```

Comment:
Breaks backwards compat it should use the old usededicatedtenancy variable instead of always passing tenancy default

Reply:
added the backward compatibility